### PR TITLE
fix(fe): password check and logout

### PR DESCRIPTION
**Pull request by rolcho**
- **strong password test on user update**
- **fix(fe): logout navigates to login page**

### CODE REVIEW

**Review by markkovari — APPROVED** [review]

File: sk0-front/src/app/profile/profile.page.ts [hunk 1]
```diff
@@ -59,6 +59,11 @@ export class ProfilePage {
   onUpdate() {
     const { name, email, password } = this.profileForm.value;
 
+    if (password && !this.strongPasswordValidator.test(password)) {
+      this.notifyWith.toastMessage("A jelszó nem elég erős", "top");
```
Comment (markkovari) [inline sk0-front/src/app/profile/profile.page.ts]: Should be awaited here, since it is an async function.

**Review by markkovari — CHANGES_REQUESTED** [review]
I spotted another `async`-`await` ambiguity.

File: sk0-front/src/app/profile/profile.page.ts (hunk 1 above)
Comment (rolcho) [reply sk0-front/src/app/profile/profile.page.ts]: hmmm... it is not clear ofr me. if we should use async-await here like:
```typescript
async onUpdate() {
  //...some code
  await this.notifyWith.toastMessage()
}
```
should we use here as well, and everywhere where we use navigation or some Promise<> return values?
```typescript
async onCancel() {
    await this.navCtl.navigateBack("/home");
  }
```

Comment (rolcho) [reply sk0-front/src/app/profile/profile.page.ts]: toasMessage returns ```Promise<void>```

Comment (markkovari) [reply sk0-front/src/app/profile/profile.page.ts]: Yes, we should at least for testability purposes.
If you think about the "side effects", these are the places where you might just "register" a function and it is not blocking at that line and executes everything else after that, which might lead to unexpected things.

Comment (markkovari) [reply sk0-front/src/app/profile/profile.page.ts]: `Promise<void>` is fine, you can (should) await it in most of the cases.
We should avoid now parallel executions and data races.

Comment (rolcho) [reply sk0-front/src/app/profile/profile.page.ts]: so in a toast message tha has a timer wont block everything else until it disappears?

Comment (markkovari) [reply sk0-front/src/app/profile/profile.page.ts]: As far as I understood from the documentation https://ionicframework.com/docs/api/alert#present
the overlay creation is the async function and not the whole display.
We are calling the `present` method, and it takes time, not the whole timeout of it.

But honestly I might be wrong.

**Review by markkovari — APPROVED** [review]

**Review by Jankita58 — APPROVED** [review]

**Review by bendesbeus — APPROVED** [review]

**Pull request merged**